### DISCUSSION TITLE
How to investigate heap out of memory in NextJS

### DISCUSSION THREAD

**Discussion by nodeworks** [vercel/next.js#17295] [Help]
Was wondering where I can dig in to get more information on this Ineffective mark-compacts heap error:

![Screen Shot 2020-09-22 at 9 44 48 PM](https://user-images.githubusercontent.com/5272855/93955218-3e572980-fd1d-11ea-9526-86947679dfa0.png)

The only thing I can think of is that I use a long maxInactive age:

```
onDemandEntries: {
  maxInactiveAge: 24 * 60 * 60000
}
```

But would that actually be the reason if I'm only really working on 1 or 2 pages? The error usually comes about after 2-3 days.

**Comment by jamesmosier** [comment]
You could maybe try increasing the heap memory size [as mentioned here](https://github.com/vercel/next.js/issues/15345#issuecomment-661411878): `node --max-old-space-size=2048 ./node_modules/.bin/next dev`

Aside from that, there are a [few other issues mentioning this](https://github.com/vercel/next.js/issues?q=is%3Aissue+out+of+memory) that may or may not be helpful.

**Reply by nodeworks** [reply]
I have max old space size set:

```--max_old_space_size=18192```

**Comment by devdogfish** [comment]
I had the same bug and after hours of digging found the error in my code. It was an infinite loop caused by a mistake in the JWT tokens and authentication flow, where a condition triggered the token function to repeat endlessly, eventually consuming all the memory. 

That is why increasing memory also didn't work.

The cause might be different for you, but just a tip: check your code for infinite loops.
